services/QuillLMS/client/app/bundles/Shared/components/studentLessons/register.tsx

```diff
@@ -75,13 +76,43 @@ class Register extends React.Component<any, any> {
     );
   }
 
+  translatedText = (): string => {
+    function getTranslation( translations, targetLanguage): string | undefined {
+      const targetLocale = Object.entries(localeToLanguageMap).find(
+        ([_, language]) => language === targetLanguage
+      )?.[0];
+
+      if (targetLocale && targetLocale in translations) {
+        return translations[targetLocale];
+      }
+
+      // If the exact locale isn't found, try to find a matching language
```

Comment:
I'm confused about why we need both of these iterations. It seems like we could just cut the first one and always rely on the second? I think I'm missing how the first one offers greater specificity since we're always just grabbing the first item returned, unless it's the ordering of the `localeToLanguageMap` that matters?

Reply:
looking at this again, I think it's better if the ruby API brings back the languages by name and the JS doesn't have to worry about all these mappings back and forth